streaming/storage/deque.go

```diff
@@ -0,0 +1,392 @@
+package storage
+
+import (
+	"github.com/cube2222/octosql"
+	"github.com/golang/protobuf/proto"
+	"github.com/pkg/errors"
+)
+
+/*
+	Name clarification:
+	In a deque 1, 2, 3, 4 the Front is the 1 and the Back is the 4. This way
+	a Push/PopFront modify the front attributes, and Push/PopBack modify the back attributes.
+*/
+const (
+	defaultFirstFreeFrontSpot = 0
+	defaultFirstFreeBackSpot  = 1
+)
+
+var (
+	dequeValueKeyPrefix     = []byte("value")
```

Comment:
I think we should stick to one prefix naming convention. I've always been using $prefix$, here it's without the dollars, let's decide on one.